crates/biome_js_analyze/src/lint/nursery/no_yoda_expression.rs

```diff
@@ -0,0 +1,473 @@
+use crate::{utils::is_node_equal, JsRuleAction};
+use biome_analyze::{
+    context::RuleContext, declare_rule, ActionCategory, Ast, FixKind, Rule, RuleDiagnostic,
+    RuleSource,
+};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_factory::make::{self, js_binary_expression, token};
+use biome_js_syntax::{
+    AnyJsExpression, AnyJsLiteralExpression, AnyJsStatement, JsBinaryExpression, JsBinaryOperator,
+    JsCallExpression, JsIfStatement, JsLanguage, JsLogicalExpression, JsLogicalOperator,
+    JsParenthesizedExpression, JsSyntaxKind, JsUnaryOperator, JsWhileStatement, JsYieldArgument,
+    JsYieldExpression, T,
+};
+use biome_rowan::{AstNode, BatchMutationExt, SyntaxTriviaPiece, TriviaPieceKind, WalkEvent};
+
+declare_rule! {
+    /// Disallow the use of yoda expressions.
+    ///
+    /// A Yoda expression is a programming style where the two parts of the expression are reversed from the typical order in a conditional statement.
+    /// A Yoda expression places the constant portion of the expression on the left side of the conditional statement. They can be confusing to some people,
+    /// the rule forbids the use of it to improve code readability.
+    ///
```

Comment:
We should say that there is an exception with ranges and point to the example section.

Reply:
forgot about that, done!